geopandas/tools/clip.py

```diff
@@ -65,14 +82,22 @@ def clip(gdf, mask, keep_geom_type=False):
     If there are multiple polygons in mask, data from `gdf` will be
     clipped to the total boundary of all polygons in mask.
 
+    If the `mask` is a tuple of `(minx, miny, maxx, maxy)`, a faster rectangle
+    clipping algorithm will be used. Note that this can lead to slightly different
+    results in edge cases, e.g. if a line would be reduced to a point, this point might
+    not be returned.
```

Comment:
Can you also add the note on validity we have in `clip_by_rect`?

Reply:
Done